mozilla/blurts-server
# MNTOR-3814 - use context to fetch experiment data from Cirrus

# References:

Jira: MNTOR-3814



# Description

This PR introduces an `ExperimentsContext` with a provider and a `useExperiments()` hook that can be called by any client component that needs it. I implemented it just for Glean so far (per the linked Jira ticket) but if this approach is OK I can file a followup to move from prop-drilling to this more generally for features that need access to experiment data (which features are activated, what the active branch is, etc)

# How to test

I've been using the Glean debug dashboard to inspect pings. Note that local pings have `-1` for most `Enrollments` values, you need to use Cirrus (or something that mocks it). I'll see if I can mock this to add some automated tests.

An easy way to inspect Glean pings is to run this in the browser devtools console:
```js
Glean.setLogPings(true)
```
Which should log:
```
Pings will be logged to the console until this tab is closed.
```
# Checklist (Definition of Done)
